internal/tpm/verify.go

```diff
@@ -0,0 +1,308 @@
+package tpm
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"fmt"
+	"math/big"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/transport"
+)
+
+//  +---------------------------+
+//  |    TPM Manufacturer       |
+//  |  (Root of Trust Anchor)   |
+//  +-------------+-------------+
+//                |
+//                v
+//         +---------------+
+//         |      EK       |
+//         |   (in TPM)    |
+//         +-------+-------+
+//                 | EK Cert (X.509)
+//                 v
+//  +---------------------------+
+//  |    Owner / Admin Domain   |
+//  +-------------+-------------+
+//                |
+//         +------v------+
+//         |     LAK     | <-- Proof of Residency
+//         +------+------+
+//                | Certify(LAK) signed by EK
+//                v
+//         +-------------+
+//         |   LDevID    | <-- Proof of Residency
+//         +------+------+
+//                | Certify(LDevID) signed by EK
+//                v
+//      +--------------------+
+//      |   CSR signed by    |
+//      | LDevID private key |
+//      +--------------------+
+
+// AttestationBundle contains the structured data required for TCG spec compliance
+// This includes the TPM2_Certify results for both LAK and LDevID keys
+type AttestationBundle struct {
+	// EKCert is the Endorsement Key certificate from the TPM manufacturer
+	EKCert []byte
+	// LAKCertifyInfo contains the TPM2_Certify result for the LAK signed by EK
+	LAKCertifyInfo []byte
+	// LAKCertifySignature is the signature over LAKCertifyInfo made by the EK
+	LAKCertifySignature []byte
+	// LDevIDCertifyInfo contains the TPM2_Certify result for the LDevID signed by EK
+	LDevIDCertifyInfo []byte
+	// LDevIDCertifySignature is the signature over LDevIDCertifyInfo made by the EK
+	LDevIDCertifySignature []byte
+	// LAKPublicKey is the public portion of the LAK
+	LAKPublicKey []byte
+	// LDevIDPublicKey is the public portion of the LDevID
+	LDevIDPublicKey []byte
+}
+
+// CertifyLAKWithEK uses TPM2_Certify to prove the LAK was created by this TPM
+// This implements §5.6, §5.3 of the TCG spec using AK for signing
+func (t *Client) CertifyLAKWithEK(qualifyingData []byte) ([]byte, []byte, error) {
+	if t.lak == nil {
+		return nil, nil, fmt.Errorf("LAK not initialized")
+	}
+
+	// use AK for signing instead of EK as per TCG spec
+	ak, err := t.getAKForSigning()
```

Comment:
Do we need this step? `client.AttestationKeyRSA` is just certifying the `client.AttestationKeyECC` at this point.